### ISSUE TITLE
Provide an option to sort the files under folders in the sidebar

### ISSUE THREAD

**Issue by assayire** [zed-industries/zed#4533]
### Check for existing issues

- [X] Completed

### Describe the feature

The files/folders under a folder in the sidebar is always sorted alphabetically. But the following would be very useful:

- Choose a different global default sort order (by type or last modified). Could be specified in the settings json.
- Say right-click a folder and change the sort order of the files but listing folders first.

This is useful when working with files listed in alphabetical order is not user friendly such as having to sift through files of different extensions but same name. Or having to locate the file you are working on actively in the list.

**Comment by JosephTLyons** [comment]
Additional info here on where to look:

https://github.com/zed-industries/zed/issues/8210#issuecomment-1960109656

**Comment by crbellis** [comment]
Hey @JosephTLyons, just created a pull request adding the dirs first option, along with a new project panel setting called `sort_order`. I can start working on of these other sorting options mentioned in this issue.

PR: [9093](https://github.com/zed-industries/zed/pull/9093)

**Comment by dahjelle** [comment]
If I'm not mistaken, the macOS Finder sorts files and folders together by default, while the Windows File Explorer usually puts directories first and then files. It looks like Zed does directories first and then files? It would be nice to default to the platform behavior and then allow user configuration.

**Comment by unikitty37** [comment]
Yes, Zed is using Windows-style sorting for me on this Mac, and it is causing no end of confusion.

While it would be nice to default to the platform behaviour, we should at least have a way to specify _which_ behaviour we want, rather than being forced to do things the Windows way.

**Comment by tudor-cel-stan** [comment]
In Sveltekit or the new App router for Next this is a huge time saver. I think we would really benefit from a simple files first config option before getting to more complex sorting.

**Comment by jianwolf** [comment]
Is it possible to reopen this issue? Our projects would benefit a lot if we have a "sort by last modified date" option. Thank you!
